extension-manifest-v3/scripts/download-engines/utils.js

```diff
@@ -55,12 +64,31 @@ export function getCompatRule(rule) {
         rule.condition.excludedInitiatorDomains ||
         rule.condition.excludedRequestDomains
       )?.map((d) => `*${d}`),
-      urlFilter:
-        rule.condition.urlFilter ||
-        (rule.condition.regexFilter ? undefined : '*'),
+      urlFilter: rule.condition.urlFilter,
       regexFilter: rule.condition.regexFilter,
+      isUrlFilterCaseSensitive: rule.condition.isUrlFilterCaseSensitive,
     },
   };
+
+  if (!newRule.condition.urlFilter && !newRule.condition.regexFilter) {
```

Comment:
The logic didn't change from the previous revision, only you added a log message. Is this gives us anything?